pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -531,8 +596,9 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 	return reconcile.Result{}, nil
 }
 
-// setDefaultsOnIntrusionDetection updates the IntrusionDetection resource with defaults if ComponentResources is not populated.
-func (r *ReconcileIntrusionDetection) setDefaultsOnIntrusionDetection(ctx context.Context, ids *operatorv1.IntrusionDetection) error {
+// fillDefaults updates the IntrusionDetection resource with defaults if
+// ComponentResources and AnomalyDetectionSpec are not populated.
+func (r *ReconcileIntrusionDetection) fillDefaults(ctx context.Context, ids *operatorv1.IntrusionDetection, isManagedCluster bool) error {
```

Comment:
isManagedCluster is no longer used in this function so you should remove it.

Reply:
removed https://github.com/tigera/operator/pull/2094/commits/e393f5a8741e3e1206575f5c92123a92ebc837f9